OSDOCS 13005: Updated release notes regarding oc-mirror v2 and v1 updates (#86618)
openshift/openshift-docs: release_notes/ocp-4-18-release-notes.adoc

[id="ocp-release-notes-openshift-cli_{context}"]
=== OpenShift CLI (oc)

> oc-mirror plugin v2 is now generally available. To use it, add the `--v2` flag when running oc-mirror commands. The previous version (oc-mirror plugin v1), which runs when the `--v2` flag is not set, is now deprecated. We encourage customers to transition to oc-mirror plugin v2 for continued support and improvements.

Reviewer: ```suggestion
oc-mirror plugin v2 is now generally available. To use it, add the `--v2` flag when running oc-mirror commands (Migration from v1 to v2 docs). The previous version (oc-mirror plugin v1), which runs when the `--v2` flag is not set, is now deprecated. We encourage customers to transition to oc-mirror plugin v2 for continued support and improvements.
```
@subhtk is it possible to add a link to our migration docs on `Migration from v1 to v2 docs` I suggested above?

Author: yeah no issues.

> oc-mirror plugin v2 now supports mirroring Helm charts and Helm repositories. Also,  oc-mirror plugin v2 can now be used in environments where HTTP/S proxy is enabled, ensuring broader compatibility with enterprise setups.

Reviewer: ```suggestion
oc-mirror plugin v2 now supports mirroring Helm charts. Also,  oc-mirror plugin v2 can now be used in environments where HTTP/S proxy is enabled, ensuring broader compatibility with enterprise setups.
```

> oc-mirror plugin v2 introduces retro-compatible filtering of operator catalogs and generates filtered catalogs. This feature allows cluster administrators to view only the operators that have been mirrored, rather than the complete list from the origin catalog.

Reviewer: I would say v1 retro-compatible filtering.

What do you think @sherine-k ?

Author: @sherine-k Thoughts on this?

Reviewer 2: yes agreed!

[id="ocp-release-notes-openshift-cli_{context}"]
=== OpenShift CLI (oc)

> oc-mirror plugin v2 is now generally available. To use it, add the `--v2` flag when running oc-mirror commands. The previous version (oc-mirror plugin v1), which runs when the `--v2` flag is not set, is now deprecated. We encourage users to transition to oc-mirror plugin v2 for continued support and improvements.

Reviewer 3: I would add a heading here specifically for this GA announcement:
```suggestion
[id="ocp-release-notes-openshift-cli-oc-mirror_{context}"]
==== oc-mirror plugin v2 (Generally Available)

oc-mirror plugin v2 is now generally available. To use it, add the `--v2` flag when running oc-mirror commands. The previous version (oc-mirror plugin v1), which runs when the `--v2` flag is not set, is now deprecated. We encourage users to transition to oc-mirror plugin v2 for continued support and improvements.
```

> oc-mirror plugin v2 is now generally available. To use it, add the `--v2` flag when running oc-mirror commands. The previous version (oc-mirror plugin v1), which runs when the `--v2` flag is not set, is now deprecated. We encourage users to transition to oc-mirror plugin v2 for continued support and improvements.

Reviewer 3: I think this sounds a little more formal in docs that using "we" imo

```suggestion
oc-mirror plugin v2 is now generally available. To use it, add the `--v2` flag when running oc-mirror commands. The previous version (oc-mirror plugin v1), which runs when the `--v2` flag is not set, is now deprecated. It is recommended to transition to oc-mirror plugin v2 for continued support and improvements.
```

oc-mirror plugin v2 now supports mirroring helm charts. Also, oc-mirror plugin v2 can now be used in environments where `HTTP/S` proxy is enabled, ensuring broader compatibility with enterprise setups.

oc-mirror plugin v2 introduces v1 retro-compatible filtering of Operator catalogs and generates filtered catalogs. This feature allows cluster administrators to view only the Operators that have been mirrored, rather than the complete list from the origin catalog.

> 

Reviewer 3: Lastly, I'd just recommend linking to the oc-mirror doc so that users can learn more:
```suggestion

For more information, see xref:../disconnected/mirroring/about-installing-oc-mirror-v2.adoc#about-installing-oc-mirror-v2[Mirroring images for a disconnected installation by using the oc-mirror plugin v2].
```

> 

Reviewer 2: do we want to mention something about the release signature config map being generated in v2 now?

If we don't have a known issue for the previous release, it probably isn't too important?

@lmzuccarelli WDYT?

Reviewer 4: @sherine-k - Agreed not too important

|====
|Feature |4.16 |4.17 |4.18

|oc-mirror plugin v1
|General Availability

> |General Availability

Reviewer: @subhtk is this a duplicated line or is it correct?

Author: it is correct.

|oc-mirror plugin v2
|Technology Preview
|Technology Preview
|Technology Preview
|General Availability 

|oc-mirror plugin v2 enclave support
|Technology Preview
|Technology Preview
|Technology Preview
|General Availability

|oc-mirror plugin v2 delete functionality
|Technology Preview
|Technology Preview
|Technology Preview

|====

[discrete]
[id="ocp-release-notes-extensions-tech-preview_{context}"]
=== Extensions Technology Preview features

// "Extensions" refers to OLMv1

.Extensions Technology Preview tracker
[cols="4,1,1,1",options="header"]
|====
|Feature |4.16 |4.17 |4.18

|{olmv1-first}
|Technology Preview

> |Technology Preview

Reviewer 3: I think this might have been accidentally deleted somehow, please add these lines back in (and still ensure that the delete functionality entry shows GA for 4.18)
